lib/client/session.go

```diff
@@ -600,7 +600,13 @@ func (ns *NodeSession) runCommand(ctx context.Context, mode types.SessionPartici
 	return ns.regularSession(ctx, chanReqCallback, func(s *tracessh.Session) error {
 		errCh := make(chan error, 1)
 		go func() {
-			errCh <- s.Run(ctx, strings.Join(cmd, " "))
+			err := s.Run(ctx, strings.Join(cmd, " "))
+			// If the session is running in a child process with --fork-after-authentication,
+			// stdin will already be closed and this error can be ignored.
+			if err != nil && strings.Contains(err.Error(), "/dev/stdin: file already closed") {
```

Comment:
Why can't we replace the actual fd 0 with a devnull?

Reply:
On Windows, we can use `SetStdHandle()` and `GetStdHandle()` to update the handle, but `GetStdHandle()` is only called on startup to set `os.Std*`. Since the eventual ssh session just uses `TeleportClient.Stdin`, we gain nothing over just setting `tc.Stdin` with a new file. And if that works for Windows, it'll work for everything else. If we  don't close stdin, there's no risk of something accidentally getting fd 0 anyway.

TL;DR we can, it's just pointless.

Reply:
It's a file or pipe or socket that will stay open past the point where the executor of the program expects it to stay open - and won't it potentially read from the terminal from the background if something weird happens like an OTP prompt or something along those lines?